demonstrations/tutorial_vqe_bond_dissociation.py

```diff
@@ -0,0 +1,389 @@
+
+##############################################################################
+# Hydrogen Exchange Reaction
+# -----------------------------
+#
+# After studying a simple diatomic bond dissociation, we move to a slightly more complicated
+# case: a hydrogen exchange reaction.
+#
+# .. math:: H_2 + H \rightarrow H + H_2.
+#
+# This reaction has a barrier, the `transition state
+# <https://en.wikipedia.org/wiki/Transition_state>`_, that must be crossed
+# for the exchange of an atom to be complete. In this case, the transition state
+# corresponds to a specific linear arrangement of the atoms where one :math:`H-H` bond is
+# partially broken and the other :math:`H-H` bond is partially formed.
+# The molecular movie ⚛️🎥 below is an illustration of the reaction trajectory --- how the distance
+# between the hydrogen atoms changes as one bond is broken and another one is formed.
+# The path along which the reaction proceeds is known as the `reaction coordinate
+# <https://en.wikipedia.org/wiki/Reaction_coordinate>`_.
+#
+# .. figure:: /demonstrations/vqe_bond_dissociation/h3_mol_movie.gif
+#   :width: 50%
+#   :align: center
+#
+# In a minimal basis like STO-3G, this system consists of three electrons in six spin
+# molecular orbitals. This translates into a six-qubit problem, and the Hartree-Fock state
+# is :math:`|111000\rangle`. As there is an unpaired
+# electron, the spin multiplicity is equal to two and needs to be specified, since it differs
+# from the default value of one.
+
+symbols = ["H", "H", "H"]
+multiplicity = 2
+basis_set = "sto-3g"
+
+
+##############################################################################
+# To build a potential energy surface for the hydrogen exchange, we fix the positions of the
+# outermost atoms, and change only the placement of the middle atom. For this circuit, we employ all
+# single and double excitation gates, which can be conveniently done with the
+# :class:`~qml.templates.AllSinglesDoubles` template. The rest of the procedure follows as before.
+
+from pennylane.templates import AllSinglesDoubles
+
+energies = []
+pes_point = 0
+
+# get all the singles and doubles excitations, amd Hartree-Fock state
+electrons = 3
+orbitals = 6
+singles, doubles = qchem.excitations(electrons, orbitals)
+hf = qml.qchem.hf_state(electrons, orbitals)
+
+def circuit(params, wires):
+    AllSinglesDoubles(params, wires, hf, singles, doubles)
+
+# loop to change reaction coordinate
+r_range = np.arange(1.0, 3.0, 0.1)
+for r in r_range:
+
+    coordinates = np.array([0.0, 0.0, 0.0, 0.0, 0.0, r, 0.0, 0.0, 4.0])
+
+    # We now specify the multiplicity
+    H, qubits = qchem.molecular_hamiltonian(symbols, coordinates, mult=multiplicity)
+
+    dev = qml.device("default.qubit", wires=qubits)
+    cost_fn = qml.ExpvalCost(circuit, H, dev)
+    opt = qml.GradientDescentOptimizer(stepsize=1.5)
+
+    params = np.zeros(len(singles) + len(doubles))
+
+    if pes_point > 1:
```

Comment:
am I missing something here? Why not to take the opt parameters already for the second configuration?
```suggestion
    if pes_point > 0:
```